end2end-test-examples/gcs/src/main/java/io/grpc/gcs/JavaClient.java

```diff
@@ -105,22 +108,36 @@ public void makeRandomMediaRequest(ResultTable results, int threadId) throws IOE
 
     String object = objectResolver.Resolve(threadId, /*objectId=*/ 0);
     BlobId blobId = BlobId.of(args.bkt, object);
-    ReadChannel reader = client.reader(blobId);
+
+    int capacity = args.buffSize * 1024;
+    ByteBuffer buff = ByteBuffer.allocate(capacity);
+
     for (int i = 0; i < args.calls; i++) {
-      long offset = (long) r.nextInt(args.size - args.buffSize) * 1024;
-      reader.seek(offset);
+      buff.clear();
+      int bound = args.size - args.buffSize;
+      long offset = r.nextInt(bound) * 1024L;
 
       long start = System.currentTimeMillis();
-      ByteBuffer buff = ByteBuffer.allocate(args.buffSize * 1024);
-      reader.read(buff);
-      long dur = System.currentTimeMillis() - start;
-      if (buff.remaining() > 0) {
-        logger.warning("Got remaining bytes: " + buff.remaining());
+      try (
+          ReadChannel reader = client.reader(blobId);
+          // we care that the bytes move, not what we will do after they move, route to /dev/null
+          WritableByteChannel out = Channels.newChannel(ByteStreams.nullOutputStream())
+      ) {
+        // set the chunkSize to that of our buffer to try and align range read size behavior
+        reader.setChunkSize(capacity);
+        // set begin offset
+        reader.seek(offset);
+        // set end offset (limit is counted from 0, not seek)
+        reader.limit(offset + capacity);
+
+        long copy = copy(reader, out, buff);
+        long dur = System.currentTimeMillis() - start;
+        if (copy != capacity) {
+          logger.warning(String.format("Unexpected number of bytes. Expected: %d but was: %d", capacity, copy));
+        }
+        results.reportResult(args.bkt, object, copy, dur);
```

Comment:
@veblush we were trying to understand if resportResult should be actual size of bytes read or the expected size? it would make more sense to us to have the actual size of bytes read compared to the static expected size.

Reply:
Having `actual size of bytes` here would make more sense.